python/cuml/dask/neighbors/nearest_neighbors.py

```diff
@@ -32,32 +30,40 @@ def _raise_import_exception():
                     " enable multiGPU support.")
 
 
+def _func_get_d(f, idx):
+    i, d = f
+    return d[idx]
+
+
+def _func_get_i(f, idx):
+    i, d = f
+    return i[idx]
+
+
 class NearestNeighbors(object):
     """
     Multi-node Multi-GPU NearestNeighbors Model.
     """
-    def __init__(self, client=None, **kwargs):
+    def __init__(self, client=None, streams_per_handle=0, verbose=False,
+                 **kwargs):
         self.client = default_client() if client is None else client
         self.kwargs = kwargs
+        self.X = None
+        self.Y = None
+        self.n_cols = 0
+        self.streams_per_handle = streams_per_handle
+        self.verbose = verbose
```

Comment:
why not assign self.n_neighbors_ explicitly here and make it a first class function argument?

Reply:
I've been proxying the kwargs to the distributed models on the workers to try and keep things simple and clean on the Dask side.

I keep going back and forth on this one and the client-side doesn't really have a need for this to be first-class. What do you think of changing the `kwargs` instance variable to something like `self.model_args` and documenting for users to look in that dictionary for the arguments that will be used to construct the underlying nearest neighbors models?